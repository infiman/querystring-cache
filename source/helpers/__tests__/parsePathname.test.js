import { parsePathname } from '../'

describe('parsePathname module', () => {
  describe('parsePathname', () => {
    test('empty invocation', () =>
      expect(() => parsePathname()).toThrowErrorMatchingSnapshot())

    test('root pathname', () => expect(parsePathname('/')).toEqual(['/']))

    test('simple pathname', () =>
      expect(parsePathname('/pathname/to/narnia')).toEqual([
        'pathname',
        'to',
        'narnia'
      ]))

    test('invalid pathname', () =>
      expect(() => parsePathname('invalid')).toThrowErrorMatchingSnapshot())
  })
})
