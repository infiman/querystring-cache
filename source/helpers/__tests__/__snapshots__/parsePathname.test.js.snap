// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsePathname module parsePathname empty invocation 1`] = `"Pathname is not valid. Expected: string! Received: [object Undefined]."`;

exports[`parsePathname module parsePathname invalid pathname 1`] = `"Pathname is not valid. It should start with '/'!"`;
